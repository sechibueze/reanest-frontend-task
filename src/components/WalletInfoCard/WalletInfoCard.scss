@import "../../assets/theme/colors.scss";
.wallet-info {
  background-color: $light-gray;
  border-radius: 0.4rem;
  display: flex;
  width: 100%;
  flex-direction: column;
  justify-content: center;
  padding: 0.5rem;
  box-shadow: 0 0 2px $light-gray;
}
.wallet-info-meta {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  font-size: 0.8rem;
  margin-bottom: 1.5rem;
}

.wallet-balance {
  display: flex;
  flex-direction: column;
  align-items: center;
  .wb-text {
    font-size: 0.8rem;
    margin: 0.25rem auto;
  }
  .wb-value {
    font-size: 1.5rem;
    font-weight: bold;
    margin-bottom: 0.25rem;
  }
}

.wallet-actiosn {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  & li {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    margin: 0.5rem;
    .wa-icon {
      background-color: $blue;
      padding: 0.6rem;
      border-radius: 50%;
      text-align: center;
      margin: auto;
    }
    .wallet-action-text {
      font-size: 0.6rem;
      font-weight: bold;
      margin: 0.3rem auto;
    }
  }
}
