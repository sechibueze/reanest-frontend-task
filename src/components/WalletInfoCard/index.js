import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import "./WalletInfoCard.scss";
const WalletInfoCard = () => {
  return (
    <div className="wallet-info">
      <div className="wallet-info-meta">
        <span> YOUR WALLET </span>
        <span> Feb 2, 2021 </span>
      </div>
      <div className="wallet-balance">
        <p className="wb-text"> Wallet Balance</p>
        <p className="wb-value"> N 14.61 </p>
      </div>
      <ul className="wallet-actions">
        <li>
          <span className="wa-icon">
            <FontAwesomeIcon icon="bars" color="white" />
          </span>
          <span className="wallet-action-text"> Withdraw</span>
        </li>
        <li>
          <span className="wa-icon">
            <FontAwesomeIcon icon="bars" color="white" />
          </span>
          <span className="wallet-action-text"> Fund wallet</span>
        </li>
        <li>
          <span className="wa-icon">
            <FontAwesomeIcon icon="bars" color="white" />
          </span>
          <span className="wallet-action-text"> Transactions</span>
        </li>
      </ul>
    </div>
  );
};

export default WalletInfoCard;
