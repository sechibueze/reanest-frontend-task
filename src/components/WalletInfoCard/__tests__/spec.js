import { shallow } from "enzyme";
import WalletInfoCard from "../index";
const featureCount = 3;
describe("<WalletInfoCard /> renders correctly", () => {
  const component = shallow(<WalletInfoCard />);
  const wrapper = component.find(".wallet-info");
  const features = component.find(".wallet-actions li");
  it("<WalletInfoCard /> matches snapshot", () => {
    expect(component).toMatchSnapshot();
  });
  it("<WalletInfoCard /> was rendered correctly", () => {
    expect(wrapper.length).toEqual(1);
  });
  it("Feature items rendered correctly", () => {
    expect(features.length).toEqual(featureCount);
  });
});
