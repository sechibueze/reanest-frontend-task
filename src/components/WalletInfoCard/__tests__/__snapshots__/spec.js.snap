// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WalletInfoCard /> renders correctly <WalletInfoCard /> matches snapshot 1`] = `ShallowWrapper {}`;
